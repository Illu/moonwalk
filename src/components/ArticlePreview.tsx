import React, { useContext } from "react";
import firebase from "react-native-firebase";
import styled from "styled-components/native";

import { openLink } from "../helpers/OpenLink";
import AppState from "../stores/AppState";

const Wrapper = styled.TouchableOpacity<{ isFirst: boolean }>`
  width: 100%;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  padding: 10px 16px;
  border-top-width: ${({ isFirst }) => (isFirst ? 0 : 1)}px;
  background: ${({ theme }) => theme.colors.secondary};
  border-color: ${({ theme }) => theme.colors.uiAccent};
`;

const Thumbnail = styled.ImageBackground`
  height: 100px;
  width: 100px;
  border-radius: 10px;
  overflow: hidden;
  margin-right: 10px;
`;

const Title = styled.Text`
  text-align: left;
  padding-bottom: 5px;
  font-weight: 700;
  color: ${({ theme }) => theme.colors.text};
`;

const Subtitle = styled.Text`
  color: ${({ theme }) => theme.colors.secondaryText};
`;

const DetailsWrapper = styled.View`
  flex: 1;
  padding: 10px 5px;
`;

const Dot = styled.View`
  width: 10px;
  height: 10px;
  border-radius: 5px;
  background: ${({ theme }) => theme.colors.accent};
`;

interface Props {
  article: any;
  timePosted: number;
  isFirst?: boolean;
}

const ArticlePreview: React.FC<Props> = ({
  article,
  timePosted,
  isFirst = false,
}) => {
  const appStateStore = useContext(AppState);

  const onArticlePress = () => {
    firebase.analytics().logEvent("OPEN_NEWS_ARTICLE", {
      title: article.title,
      site: article.newsSite,
    });
    openLink(article.url, appStateStore.browser);
  };

  return (
    <Wrapper onPress={onArticlePress} isFirst={isFirst}>
      <Thumbnail source={{ uri: article.imageUrl }} />
      <DetailsWrapper>
        <Title>{article.title}</Title>
        <Subtitle>
          {article.newsSite} <Dot /> {timePosted}
        </Subtitle>
      </DetailsWrapper>
    </Wrapper>
  );
};

export default ArticlePreview;
