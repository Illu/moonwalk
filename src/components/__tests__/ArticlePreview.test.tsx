import React from "react";

import TestRenderer from "../../helpers/testRenderer";
import ArticlePreview from "../ArticlePreview";

jest.mock("../../helpers/OpenLink", () => ({openLink: jest.fn()}))

describe("ArticlePreview", () => {
  it("renders correctly", () => {
    const tree = TestRenderer(
      <ArticlePreview article={{}} timePosted={0} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly at the top of the list", () => {
    const tree = TestRenderer(
      <ArticlePreview article={{}} timePosted={0} isFirst={true} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
