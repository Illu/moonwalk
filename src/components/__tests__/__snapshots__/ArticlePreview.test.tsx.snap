// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticlePreview renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "borderColor": "#c3c4c6",
      "borderTopWidth": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 1,
      "paddingBottom": 10,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 10,
      "width": "100%",
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Array [
        Object {
          "borderRadius": 10,
          "height": 100,
          "marginRight": 10,
          "overflow": "hidden",
          "width": 100,
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "uri": undefined,
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": undefined,
          },
          undefined,
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "paddingBottom": 10,
          "paddingLeft": 5,
          "paddingRight": 5,
          "paddingTop": 10,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontWeight": "700",
            "paddingBottom": 5,
            "textAlign": "left",
          },
        ]
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": "#666",
          },
        ]
      }
    >
       
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#0a84ff",
              "borderRadius": 5,
              "height": 10,
              "width": 10,
            },
          ]
        }
      />
       
      0
    </Text>
  </View>
</View>
`;

exports[`ArticlePreview renders correctly at the top of the list 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "borderColor": "#c3c4c6",
      "borderTopWidth": 0,
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 1,
      "paddingBottom": 10,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 10,
      "width": "100%",
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Array [
        Object {
          "borderRadius": 10,
          "height": 100,
          "marginRight": 10,
          "overflow": "hidden",
          "width": 100,
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "uri": undefined,
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": undefined,
          },
          undefined,
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "paddingBottom": 10,
          "paddingLeft": 5,
          "paddingRight": 5,
          "paddingTop": 10,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontWeight": "700",
            "paddingBottom": 5,
            "textAlign": "left",
          },
        ]
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": "#666",
          },
        ]
      }
    >
       
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#0a84ff",
              "borderRadius": 5,
              "height": 10,
              "width": 10,
            },
          ]
        }
      />
       
      0
    </Text>
  </View>
</View>
`;
